test/data_frame_test.exs

```diff
@@ -2,7 +2,76 @@ defmodule DataFrameTest do
   use ExUnit.Case
   doctest DataFrame
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  alias DataFrame.Frame
+
+  def empty_frame do
+    %Frame{values: [[]], index: [], columns: []}
+  end
+
+  def single_entry_frame do
+    DataFrame.new([[0, 0]], [:A])
+  end
+
+  describe "new/2" do
+    test "empty input is an empty Frame" do
+      # should the index be [[]]? Shouldn't it be []?
+      assert DataFrame.new([[]], []) == 
+        %Frame{values: [[]], index: [[]], columns: []}
+    end
+
+    test "a regular dataframe" do
+      # the index seems a bit counter-intuitive. Shouldn't this just be [0, 1]?
+      assert DataFrame.new([[0, 0], [1, 1]], [:A]) ==
+        %Frame{values: [[0], [1]], index: [[0], [1]], columns: [:A]}
+    end
```

Comment:
you are absolutely right, I've solved this in master. See my comment above, I think that would make sense to you now.